### PR TITLE
milestone to be merged to maste

### PR DESCRIPTION
**Pull request by kbjude**

### CODE REVIEW

**Review by Ispirett — CHANGES_REQUESTED** [review]
# Status changes required
Hey @kbjude!  great job so far you are doing well :+1: 

There is some bugs :bug:   to take care of please see below.

- [ ]  Unless I am a friend with a user i should not be able to see their post.

- [ ]  When I click on all users while logged in i am greeted with this error.

`ArgumentError in UsersController#index
Unsupported argument type: true (TrueClass)`

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (mosaaleb) [inline spec/models/user_spec.rb:3]: - [ ] All associations, validations, and methods in the user model need to be tested.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    end
+
+    it 'Friendhip created when there is second user' do
+      user = User.create(name: 'Jude', email: 'kk@gmail.com', password: 'Heaven')
+      user2 = User.create(name: 'Kajura', email: 'kbb@gmail.com', password: 'Heaven')
+      user.friendships.create(user_id: user.id, friend_id: user2.id, confirmed: true)
+      expect(user.friends.last).to eql(user2)
+    end
+    # it 'User is redirected to the index page' do
+    #   visit 'users/new'
```
Comment (mosaaleb) [inline spec/models/user_spec.rb]: - [ ] Comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history (screenshot from Pathwright)

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Well done @kbjude :clap:

Nice job adding friendships feature in your models, but this milestone is incomplete. We need to add the ability to "send/receive" "accept/reject" requests to accept this milestone. 

Most requirements depend on adding friendships_controller and its actions and handling the views part (friendship buttons) send/accept/reject.

- [ ] User can invite friends



- [ ] User can see friendships requests



- [ ] There is a button to invite the user to friendship on users list page and on single-user page




- [ ] Remember to add integration tests for friendships (sending, accepting, rejecting) with Capybara.



- [ ] Add a short description for this pull request summarizing what's done.

Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello! Good job... :+1:, a few changes are still required:**
:point_right: You should add a brief description to your PR
:point_right: After confirming a friendship, I could still send back another friend request and I'll have duplicate records in the db. This shouldn't happen.
:point_right: I can't access the single user page of other users when I'm logged in. It throws an error when I try to visit the url. Also, you should make user's names in the all users page link to their show page.
:point_right: You should organise your spec files properly into models and feature folders and add integration tests for friendship feature.
:point_right:  I can't find friend's posts on my timeline. My timeline should display both my posts and my friends' posts.


_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @kbjude :lollipop: 

### Thank you for working on the changes requested by the previous reviewers :medal_sports: your project is working better with each iteration so you should feel very proud of your progress, :bar_chart:  please address to the following changes

#### Milestone 5

* [ ] There is a button to invite the user to friendship on users list page and on the single-user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- The app only lets users invite other users to friendship on the list page but not on the single user page, let's work on this.

* [ ] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- The app only shows the logged-in user posts but not their friends, let's also make this possible

#### General 

- [ ] Pull Request has meaningful title and a short description ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

- This is usually a reason to invalidate a project so please make sure to add a short description in your pull request for your next submission and future projects (we cannot approve a project without it)

### That's it! :raised_hands: 

### Please :pray: make the changes :technologist: and submit for a code review when you are done :memo: 

Happy coding! :rocket: 

Any question? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Comment by kbjude** [comment]
All feedback from the TSE has been implemented.
- Having a message in to explain the PR
- Button appearing on the show page
- Being able to view messages of the friend by the logged-in user

**Comment by uryelah** [comment]
Hi **Jude Kajura Balinda**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with all the fixes from the last reviews!

About the PR message, next time write it at the top of the PR:
Screenshot_2020-06-24 milestone to be merged to maste by kbjude · Pull Request #2 · kbjude ror-social-scaffold
That's the place that should have the description. 

Also, remember to keep commits best practices next milestone:
- Capitalize your commits
- Use the imperative/present tense, not the past tense

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being an awesome dev and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**